libutils/mutex.c

```diff
@@ -69,3 +68,53 @@ void __ThreadUnlock(pthread_mutex_t *mutex,
         exit(101);
     }
 }
+
+int __ThreadWait(pthread_cond_t *pcond, pthread_mutex_t *mutex, int timeout,
```

Comment:
Please add a documentation comment to this function and refer to the two important values of `timeout`.